compaction/task_manager_module.cc

```diff
@@ -289,6 +289,9 @@ future<> shard_major_keyspace_compaction_task_impl::run() {
     tasks::task_manager::task_ptr current_task;
     tasks::task_info parent_info{_status.id, _status.shard};
     std::vector<table_tasks_info> table_tasks;
+
+    co_await _db.flush_all_tables();
+
```

Comment:
Let's protect it with a comment, and with a configuration variable in case it causes a regression and we need to disable it in the field.